app/views/foreman_bootdisk/host.erb

```diff
@@ -7,9 +7,23 @@
 <%
 interface = @host.provision_interface
 bootdisk_raise(N_('Host has no provisioning interface defined')) unless interface
-bootdisk_raise(N_('Host has no IP address defined')) if interface.ip.nil? || interface.ip.empty?
-bootdisk_raise(N_('Host has no subnet defined')) unless interface.subnet
+bootdisk_raise(N_('Host has no IPv4 or IPv6 address defined')) unless interface.ip.present? || interface.ip6.present?
+bootdisk_raise(N_('Host has no subnet defined')) unless interface.subnet || interface.subnet6
 bootdisk_raise(N_('Host has no domain defined')) unless interface.domain
+if interface.ip.present? && interface.subnet
+  ip = interface.ip
+  mask = interface.subnet.mask
+  gw = interface.subnet.gateway
+  dns = interface.subnet.dns_servers.first
+elsif interface.ip6.present? && interface.subnet6
+  ip = interface.ip6
+  mask = interface.subnet6.mask
+  gw = interface.subnet6.gateway
+  dns = interface.subnet6.dns_primary || interface.subnet6.dns_secondary
```

Comment:
You should be able to use `subnet6.dns_servers.first` here as well.

How about
```ruby
if interface.ip.present? && interface.subnet
  subnet = interface.subnet
elsif interface.ip6.present? && interface.subnet6
  subnet = interface.subnet6
end

subnet.mask
subnet.dns_servers.first
subnet.gateway
```

Reply:
For some reason I could not find, it would not work. Safemode was the issue, I could not understand it because we allow this method for Subnet and IPv6 is not a subclass, is it?

Reply:
We have `Subnet::Ipv4` and `Subnet::Ipv6`.

Reply:
Oh I think I know - I was testing this out on Foreman 1.20 and it might not have this method in safemode allowed methods.

Reply:
Then it's not available for both IPv4 and IPv6.